internal/service/kms/secrets_data_source.go

```diff
@@ -21,24 +22,33 @@ func DataSourceSecrets() *schema.Resource {
 				Required: true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"name": {
-							Type:     schema.TypeString,
-							Required: true,
-						},
-						"payload": {
-							Type:     schema.TypeString,
-							Required: true,
-						},
 						"context": {
 							Type:     schema.TypeMap,
 							Optional: true,
 							Elem:     &schema.Schema{Type: schema.TypeString},
 						},
+						"encryption_algorithm": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice(kms.EncryptionAlgorithmSpec_Values(), false),
+						},
 						"grant_tokens": {
 							Type:     schema.TypeList,
 							Optional: true,
 							Elem:     &schema.Schema{Type: schema.TypeString},
 						},
+						"key_id": {
+							Type:     schema.TypeString,
+							Optional: true,
```

Comment:
Would it be possible to make this dependent on the encryption algorithm. Meaning if key is asymmetric then key_id is required

Reply:
The default algo is SYMMETRIC_DEFAULT and the key_id is retrieved from the kms meta data when the default is applied. Now to make the change where if encryption algo is not SYMMETRIC_DEFAULT, then check for keyid is already handled by the API.